content/posts/2023-03-13-Hunt-for-and-craft-your-own-malicious-usb/index.mdx

```diff
@@ -0,0 +1,115 @@
+
+# The basics of crafting a malicious USB drive
+
+There are several ways to get your hand on a malicious USB, the most popular is by far the Rubber Ducky by Hak5. The last few years there has been an increase in malicious cables, at least from my observations. The cable acts like a rubber ducky but looks like your normal USB charger cable. Another approach is to create one yourself using other popular microcontrollers. There are certain requirements for this to work but aiming for a Leonardo you are already there in terms of chipset that has this emulation feature.
+
+Way back when I started researching the topics of malicious USB's i made one myself that was able to be used regardless of the OS I wanted to attack. To achieve this, it had a few buttons, 3x2. Where the first row of buttons was for OS (Windows, MacOS and Linux), the next row was for filetypes to extract (text/documents, videos and pictures). Pressing a combination of OS button and filetype, files were then sent to a USB storage located in the same hub as the Arduino. 
+
+![DIY malicious USB device - fits unnoticed into the bookshelf.](./media/FRONTPIC.jpg)
+
+
+And here are a few of the different devices in my collection, for your reference.
+![These devices may take on any shape to get through security and to act normal!](./media/20230315_115058.jpg)
+
+# Let's go hunting
+
+Microsoft has made it easy to hunt for USBs in Sentinel using KQL. Entering into the ```DeviceEvents``` table, we get access to a lot of information ranging through most of the action types that might lead us to detecting a malicious USB device. Some of which are:
+- PnpDeviceConnected
+- PnPDeviceAllowed
+- UsbDriveMounted
+- UsbDriveUnmounted
+- UsbDriveLetterChanged
+
+But making it easier, does not equal to easy. Attackers always look for ways to hide their steps, and by changing and spoofing the USB PID and VID (Product ID and Vendor ID) they make it harder to be detected. But this is by far the most effective way also, as this means that there is an additional step in creating such a malicious USB. I finally resorted into this approach as I tried quite a few different approaches into detecting the few scripts I was running on my device. First of all, I tried to look at the ```UsbDriveMounted``` and set a timegap between that and ```UsbDriveUnmounted``` to look for all process creations and PowerShell commands created during that connection time. But as you may think, that was ineffiecient. It yielded a result not worth my time to dig through. I tried a few variations of this before I saw that I need to filter on known or potentially malicious VIDs and PIDs. And then I found a great resource to go look for such devices based on what kinds of microchip they have. Not all microchips have the ability to emulate a keyboard - those who do, may be used in these types of attacks. Here is a draft: 
```

Comment:
"But this is by far the most effective way also,"

"this" - replace "with hunting for these"

Add a potential answer to limit the endless amount of results from the tested query where you filter on known powershell modules. Maybe a link to such as well (one by Alex Verboon i think)